### ISSUE TITLE
Semantic analysis for caller capabilities

### ISSUE THREAD

**Issue by franklinsch** [flintlang/flint#2]
We need to check the validity of a function call in terms of the capabilities which are associated with it, and the context's (enclosing function) capability-level.

For
```
Foo :: (any) {
  func foo() {
    bar()
  }
}

Foo :: (owner) {
  func bar() {}
}
```
The compiler should throw an error for line 3.
